File: examples/resource/resource.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.S3()
      .allow()
      .allActions()
      .onBucket('example-bucket')
      .onObject('example-bucket', 'some/path/*')
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);

File: examples/resource/resource.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.S3() \
    .allow() \
    .all_actions() \
    .on_bucket('example-bucket') \
    .on_object('example-bucket', 'some/path/*')
    # doc-end
)

all = [s]
out(all)
deploy(all)
